WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageViewModel.swift

```diff
@@ -147,18 +156,34 @@ private extension AddProductFromImageViewModel {
             do {
                 let texts = try await imageTextScanner.scanText(from: image)
                 analytics.track(event: .AddProductFromImage.scanCompleted(source: addProductSource, scannedTextCount: texts.count))
+
+                guard texts.isNotEmpty else {
+                    throw ScanError.noTextDetected
+                }
+
                 scannedTexts = texts.map { .init(text: $0, isSelected: true) }
                 [nameViewModel, descriptionViewModel].forEach { $0.reset() }
                 generateProductDetails()
             } catch {
-                analytics.track(event: .AddProductFromImage.scanFailed(source: addProductSource, error: error))
-                DDLogError("⛔️ Error scanning text from image: \(error)")
+                switch error {
+                case ScanError.noTextDetected:
+                    if scannedTexts.isEmpty {
```

Comment:
Should we remove this check and replace the `texts.isNotEmpty` check on line 160  with it?

Reply:
This check ensures that we don't show an error when we already have a few scanned texts from a previous successful scan. (could be from a different image)

Reply:
@selanthiraiyan Sorry I don't understand your explanation. We are supposed to clear the scanned texts and error message from the previous image right? Why can it affect the next scan?

Reply:
@itsmeichigo Sorry, let me try to explain. Kindly let me know if you have any questions. 

> We are supposed to clear the scanned texts and error message from the previous image right? Why can it affect the next scan?

- The previous code before this implementation wasn't clearing the scanned texts from the previous image in case the new scan failed for some reason. 
- I hope it is because it is better to keep the `scannedTexts` of the previous image as it is probably of the same product. 
- i.e. The user took a photo from a different angle of the same product packaging to achieve better results or to add more items to the scanned texts list.

Kindly let me know your thoughts.

Reply:
I'd prefer to clear the scanned text after getting a new image to avoid weird edge cases. When the user takes a new picture, we expect a clean state, so everything should be reset.

Reply:
Logged a subtask here #10293